src/dsp_tools/xmllib/models/resource.py

```diff
@@ -1173,3 +1182,11 @@ def add_migration_metadata(
             )
         self.migration_metadata = MigrationMetadata(creation_date=creation_date, iri=iri, ark=ark, res_id=self.res_id)
         return self
+
+
+def _check_richtext_before_conversion(value: Any, res_id: str, prop_name: str) -> None:
```

Comment:
I don't want to make this a public function, which is why I did not add it to the value checkers.